src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py

```diff
@@ -340,3 +315,115 @@ def _resolve_api_version(client, provider_namespace, resource_type, parent_path)
     raise RequiredArgumentMissingError(
         'API version is required and could not be resolved for resource {}'
         .format(resource_type))
+
+
+def run_subprocess(command, stdout_show=None):
+    if stdout_show:
+        process = subprocess.Popen(command, shell=True)
+    else:
+        process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    process.wait()
+    if process.returncode:
+        logger.warning(process.stderr.read().strip().decode('UTF-8'))
+
+
+def run_subprocess_get_output(command):
+    commands = command.split()
+    process = subprocess.Popen(commands, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    process.wait()
+    return process
+
+
+def register_credential_secrets(cmd, server, repository):
+    resource_group = parse_resource_id(server.id)["resource_group"]
+    scope = "/subscriptions/{}/resourceGroups/{}".format(get_subscription_id(cmd.cli_ctx), resource_group)
+
+    app = create_service_principal_for_rbac(cmd, name=server.name, role='contributor', scopes=[scope])
+    app['clientId'], app['clientSecret'], app['tenantId'] = app.pop('appId'), app.pop('password'), app.pop('tenant')
+    app['subscriptionId'] = get_subscription_id(cmd.cli_ctx)
+    app.pop('displayName')
+    app.pop('name')
+
+    app_key_val = []
+    for key, val in app.items():
+        app_key_val.append('"{}": "{}"'.format(key, val))
+
+    app_json = ',\n  '.join(app_key_val)
+    app_json = '{\n  ' + app_json + '\n}'
```

Comment:
How about using `run_subprocess('az ad sp create-for-rbac ...')` instead of calling `create_service_principal_for_rbac` and parsing results to json by yourself?

Reply:
We need to make a credential from this command output which will require parsing anyway.. Do you still think it's necessary? 

Thanks for the comment. I'm working on introducing a new feature to allow RDBMS server users to run sql file even for private access server. This will be workaround before that feature is available so looks like we are running into lots of limitations here.